src/main.rs

```diff
@@ -772,6 +793,13 @@ where
         Commands::Config(config_args) => return config(config_args),
         Commands::Dump(dump_args) => return dump(dump_args),
         Commands::Version => return print_version(),
+        Commands::IsToprepo => {
+            let maybe = is_toprepo()?;
+            return match maybe {
+                true => Ok(()),
+                false => Err(ExitSilently.into()),
+            };
```

Comment:
What about the following?
```suggestion
            return Ok(match is_toprepo()? {
                true => ExitCode::SUCCESS,
                false => ExitCode::FAILURE,
            });
```